openwisp_users/base/models.py

```diff
@@ -221,6 +221,19 @@ def clean(self):
                 )
             )
 
+    @property
+    def name(self):
+        """
+        Returns the connected user's full name or user's username if
+        full name method is unavailable (e.g. on a custom user class)
+        or if full name is an empty string.
+        """
+        try:
+            assert self.user.get_full_name() != ''
+            return self.user.get_full_name()
+        except (AssertionError, AttributeError):
+            return str(self.user.username)
```

Comment:
why not simply:

```python
return self.user.get_full_name() or str(self.user.username)
```

Reply:
I adapted this code from django-organizations. I will update it.